vis_gen.py

```diff
@@ -0,0 +1,565 @@
+    def font_style(self):
+        return self.__font_style
+
+    @property
+    def antialias(self):
+        return self.__inv_antialias_map[self.__antialias]
+
+    @property
+    def out_dir(self):
+        return self.__out_dir
+
+    @property
+    def grayscale(self):
+        return self.__grayscale
+
+
+    @font_size.setter
+    def font_size(self, font_size):
+        """Check if font_size is 
+            1. larger than 0
+            2. smaller image_size"""
+        if font_size <= 0:
+            raise ValueError('Expect font_size to be larger than 0.')
+        elif font_size > self.image_size:
+            raise ValueError('Expect font_size to be smaller than image_size.')
+        else:
+            self.__font_size = font_size
+
+    @image_size.setter
+    def image_size(self, image_size):
+        """Check if image_size is larger than 0."""
+        if image_size <= 0:
+            raise ValueError('Expect image_size to be larger than 0.')
+        else:
+            self.__image_size = image_size
+    
+    @font_name.setter
+    def font_name(self, font_name):
+        """Check if font_name exists, if so, change to new font_name."""
+        if hasattr(self, '__font_name'):
+            full_font_name = font_name + ':style=' + self.font_style
+        else:
+            full_font_name = font_name
+        ft_face = self.__ft.find_face(full_font_name) # temporary
+        if ft_face.family_name != font_name.split(':')[0]:
+            raise ValueError("Font {} cannot be found.".format(font_name))
+        else: 
+            self.__font_name = font_name
+            self.__font_face = qah.FontFace.create_for_ft_face(ft_face)
+
+    @font_style.setter
+    def font_style(self, font_style):
+        """Add style parameter in in freetype font name. No checking if the
+        style exists."""
+        # Do nothing if font_style is None or is an empty string
+        if not font_style:
+            self.__font_style = ""
+            return
+        # Get freetype font and then create Cairo font
+        full_font_name = self.font_name.split(':')[0] + ':style=' + font_style
+        ft_face = self.__ft.find_face(full_font_name)  # temporary
+        self.__font_style = font_style
+        self.__font_face = qah.FontFace.create_for_ft_face(ft_face)
+
+    @antialias.setter
+    def antialias(self, antialias):
+        if antialias not in self.__antialias_map.keys():
+            raise ValueError('Expect antialias to be one of "Default", "Best", '
+                             '"Fast", "Good" or "None".')
+        self.__antialias = self.__antialias_map[antialias]
+
+    @out_dir.setter
+    def out_dir(self, out_dir):
+        self.__out_dir = out_dir
+
+    @grayscale.setter
+    def grayscale(self, grayscale):
+        self.__grayscale = grayscale
+        # Configure format and color according to grayscale option
+        if grayscale:
+            self.__cairo_format = CAIRO.FORMAT_A8 # 8 bits grayscale format
```

Comment:
Should  we also have default value of these in __init__ ?

Reply:
I have default value for grayscale boolean value in init. Each time self.grayscale is assigned some value, even in init, this function is called, so we know for sure these private field are initialized. The intention for not including was to avoid requiring user to import CAIRO to use this object. And yes you are right, this is a trade of between flexibility and simplicity.